modules/distribution/product/src/main/conf/synapse-configs/default/sequences/_cors_request_handler_.xml

```diff
@@ -25,6 +25,11 @@
                </filter>
             </then>
          </filter>
+         <filter source="boolean($trp:Access-Control-Expose-Headers)" regex="false">
+            <then>
+               <property name="Access-Control-Expose-Headers" expression="$ctx:Access-Control-Allow-Headers" scope="transport" type="STRING"/>
```

Comment:
@HiranyaKavishani shouldn't this be `expression="$ctx:Access-Control-Expose-Headers"`?

Reply:
Oh yes. thanx. That should be (expression="$ctx:Access-Control-Expose-Headers")

Reply:
Please fix in all repos.

Reply:
Fixed in 48efcfd3c09ef35e9b9d591989604d5c876d89dc

Reply:
How can the vale be added to ?
in version 3.1.0, api-manager.xml is not allowed to add this directly. 

<CORSConfiguration>
       .....
        <!-- Configure Access-Control-Expose-Headers -->
        <Access-Control-Expose-Headers>Location</Access-Control-Expose-Headers>

Couldn't find the notation for this in deployment.toml.

Tried this but it didn't work.

[apim.cors]
expose_headers = ["Location"]

or
[apim.cors]
access_control_expose_headers = ["Location"]

Reply:
It seems

Reply:
We have to fix this. @bhathiya Are we good to add this to WUM?

Reply:
Please create a Git issue so that we can fix it for 3.2.0

Reply:
Thanks for quick finding the reason!
I still wonder which parameter notation is correct, 'expose_headers' or 'access_control_expose_headers' in [apim.cors].

> It seems "access_control_expose_headers" cors has not been included at api-manager.xml.j2 file. It only contains the following four configurations.
> 
> ```
> <CORSConfiguration>
>     <!-- Configuration to enable/disable sending CORS headers from the Gateway-->
>     <Enabled>{{apim.cors.enable}}</Enabled>
> 
>     <!-- The value of the Access-Control-Allow-Origin header. Default values are
>          API Developer Portal addresses, which are needed for a swagger to function. -->
>     <Access-Control-Allow-Origin>{% for origin in apim.cors.allow_origins %}{{origin}}{{ "," if not loop.last }}{% endfor %}</Access-Control-Allow-Origin>
> 
>     <!-- Configure Access-Control-Allow-Methods -->
>     <Access-Control-Allow-Methods>{% for method in apim.cors.allow_methods %}{{method}}{{ "," if not loop.last }}{% endfor %}</Access-Control-Allow-Methods>
> 
>     <!-- Configure Access-Control-Allow-Headers -->
>     <Access-Control-Allow-Headers>{% for header in apim.cors.allow_headers %}{{header}}{{ "," if not loop.last }}{% endfor %}</Access-Control-Allow-Headers>
> 
>     <!-- Configure Access-Control-Allow-Credentials -->
>     <!-- Specifying this header to true means that the server allows cookies (or other user credentials) to be included on cross-origin requests.
>          It is false by default and if you set it to true then make sure that the Access-Control-Allow-Origin header does not contain the wildcard (*) -->
>     <Access-Control-Allow-Credentials>{{apim.cors.allow_credentials}}</Access-Control-Allow-Credentials>
> </CORSConfiguration>
> ```